pgvectorscale/src/util/mod.rs

```diff
@@ -48,7 +48,7 @@ pub struct ReadableBuffer<'a> {
 }
 
 impl<'a> ReadableBuffer<'a> {
-    pub fn get_data_slice(&self) -> &[u8] {
+    pub fn get_data_slice(&self) -> &'a [u8] {
```

Comment:
does this detach the lifetime or the slice from the lifetime of self? That seems dangerous